addons/mail/wizard/mail_compose_message.py

```diff
@@ -445,6 +445,13 @@ def get_mail_values(self, res_ids):
                     {'model': 'mail.message', 'res_id': 0}
                 )['attachment_ids']
 
+            # generate message_id directly; instead of letting mail_message create
+            # method doing it. Then use it to craft references, allowing to keep
+            # a trace of message_id even when email providers override it
+            message_id = self.env['mail.message']._get_message_id(mail_values)
+            mail_values['message_id'] = message_id
+            mail_values['references'] = message_id
```

Comment:
isn't it gonna rewrite the previous references we had? like if we are replying to an email?

Reply:
Composer is used only for outgoing emails, in mailing mode, we don't reply. It is always used as a "new mailing on records". Aka: we never set a parent_id on those messages, even if they might appear on a document with existing messages.